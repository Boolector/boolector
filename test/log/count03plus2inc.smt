(benchmark count3inc
:logic QF_BV
:extrafuns ((s0 BitVec[3]))
:extrafuns ((zero BitVec[3]))
:extrafuns ((one BitVec[3]))
:extrafuns ((goal BitVec[3]))
:extrafuns ((lim BitVec[3]))
:assumption (= goal bv7[3])
:assumption (= zero bv0[3])
:assumption (= one bv1[3])
:assumption (= lim bv2[3])
:assumption (= s0 zero)
:formula (= s0 goal)
:extrafuns ((r1 Bool))
:extrafuns ((e1 Bool))
:extrafuns ((s1 BitVec[3]))
:extrafuns ((i1 BitVec[3]))
:assumption (bvule i1 lim)
:assumption (= s1 (ite r1 zero (ite e1 (bvadd s0 i1) s0)))
:formula (= s1 goal)
:extrafuns ((r2 Bool))
:extrafuns ((e2 Bool))
:extrafuns ((s2 BitVec[3]))
:extrafuns ((i2 BitVec[3]))
:assumption (bvule i2 lim)
:assumption (= s2 (ite r2 zero (ite e2 (bvadd s1 i2) s1)))
:formula (= s2 goal)
:extrafuns ((r3 Bool))
:extrafuns ((e3 Bool))
:extrafuns ((s3 BitVec[3]))
:extrafuns ((i3 BitVec[3]))
:assumption (bvule i3 lim)
:assumption (= s3 (ite r3 zero (ite e3 (bvadd s2 i3) s2)))
:formula (= s3 goal)
:extrafuns ((r4 Bool))
:extrafuns ((e4 Bool))
:extrafuns ((s4 BitVec[3]))
:extrafuns ((i4 BitVec[3]))
:assumption (bvule i4 lim)
:assumption (= s4 (ite r4 zero (ite e4 (bvadd s3 i4) s3)))
:formula (= s4 goal)
:extrafuns ((r5 Bool))
:extrafuns ((e5 Bool))
:extrafuns ((s5 BitVec[3]))
:extrafuns ((i5 BitVec[3]))
:assumption (bvule i5 lim)
:assumption (= s5 (ite r5 zero (ite e5 (bvadd s4 i5) s4)))
:formula (= s5 goal)
:extrafuns ((r6 Bool))
:extrafuns ((e6 Bool))
:extrafuns ((s6 BitVec[3]))
:extrafuns ((i6 BitVec[3]))
:assumption (bvule i6 lim)
:assumption (= s6 (ite r6 zero (ite e6 (bvadd s5 i6) s5)))
:formula (= s6 goal)
:extrafuns ((r7 Bool))
:extrafuns ((e7 Bool))
:extrafuns ((s7 BitVec[3]))
:extrafuns ((i7 BitVec[3]))
:assumption (bvule i7 lim)
:assumption (= s7 (ite r7 zero (ite e7 (bvadd s6 i7) s6)))
:formula (= s7 goal)
)
