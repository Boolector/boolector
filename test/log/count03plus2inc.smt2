(set-logic QF_BV)
(set-option :incremental true)
(declare-fun s0 () (_ BitVec 3))
(declare-fun zero () (_ BitVec 3))
(declare-fun one () (_ BitVec 3))
(declare-fun goal () (_ BitVec 3))
(declare-fun lim () (_ BitVec 3))
(declare-fun r1 () (_ BitVec 1))
(declare-fun e1 () (_ BitVec 1))
(declare-fun s1 () (_ BitVec 3))
(declare-fun i1 () (_ BitVec 3))
(declare-fun r2 () (_ BitVec 1))
(declare-fun e2 () (_ BitVec 1))
(declare-fun s2 () (_ BitVec 3))
(declare-fun i2 () (_ BitVec 3))
(declare-fun r3 () (_ BitVec 1))
(declare-fun e3 () (_ BitVec 1))
(declare-fun s3 () (_ BitVec 3))
(declare-fun i3 () (_ BitVec 3))
(declare-fun r4 () (_ BitVec 1))
(declare-fun e4 () (_ BitVec 1))
(declare-fun s4 () (_ BitVec 3))
(declare-fun i4 () (_ BitVec 3))
(declare-fun r5 () (_ BitVec 1))
(declare-fun e5 () (_ BitVec 1))
(declare-fun s5 () (_ BitVec 3))
(declare-fun i5 () (_ BitVec 3))
(declare-fun r6 () (_ BitVec 1))
(declare-fun e6 () (_ BitVec 1))
(declare-fun s6 () (_ BitVec 3))
(declare-fun i6 () (_ BitVec 3))
(declare-fun r7 () (_ BitVec 1))
(declare-fun e7 () (_ BitVec 1))
(declare-fun s7 () (_ BitVec 3))
(declare-fun i7 () (_ BitVec 3))
(assert (= goal (_ bv7 3)))
(assert (= zero (_ bv0 3)))
(assert (= one (_ bv1 3)))
(assert (= lim (_ bv2 3)))
(assert (= s0 zero))
(push 1)
(assert (= s0 goal))
(check-sat)
(pop 1)

(assert (bvule i1 lim))
(assert (= s1 (ite (= r1 (_ bv1 1)) zero (ite (= e1 (_ bv1 1)) (bvadd s0 i1) s0))))
(push 1)
(assert (= s1 goal))
(check-sat)
(pop 1)

(assert (bvule i2 lim))
(assert (= s2 (ite (= r2 (_ bv1 1)) zero (ite (= e2 (_ bv1 1)) (bvadd s1 i2) s1))))
(push 1)
(assert (= s2 goal))
(check-sat)
(pop 1)

(assert (bvule i3 lim))
(assert (= s3 (ite (= r3 (_ bv1 1)) zero (ite (= e3 (_ bv1 1)) (bvadd s2 i3) s2))))
(push 1)
(assert (= s3 goal))
(check-sat)
(pop 1)

(assert (bvule i4 lim))
(assert (= s4 (ite (= r4 (_ bv1 1)) zero (ite (= e4 (_ bv1 1)) (bvadd s3 i4) s3))))
(push 1)
(assert (= s4 goal))
(check-sat)
(pop 1)

(assert (bvule i5 lim))
(assert (= s5 (ite (= r5 (_ bv1 1)) zero (ite (= e5 (_ bv1 1)) (bvadd s4 i5) s4))))
(push 1)
(assert (= s5 goal))
(check-sat)
(pop 1)

(assert (bvule i6 lim))
(assert (= s6 (ite (= r6 (_ bv1 1)) zero (ite (= e6 (_ bv1 1)) (bvadd s5 i6) s5))))
(push 1)
(assert (= s6 goal))
(check-sat)
(pop 1)

(assert (bvule i7 lim))
(assert (= s7 (ite (= r7 (_ bv1 1)) zero (ite (= e7 (_ bv1 1)) (bvadd s6 i7) s6))))
(push 1)
(assert (= s7 goal))
(check-sat)
(pop 1)

