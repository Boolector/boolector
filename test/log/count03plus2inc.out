unsat
unsat
unsat
unsat
sat
sat
sat
sat
